src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -0,0 +1,240 @@
+import { Request, RequestHandler } from 'express'
+
+import { createLoggerWithLabel } from '../../../../config/logger'
+import { AuthType, FieldResponse } from '../../../../types'
+import { CaptchaFactory } from '../../../services/captcha/captcha.factory'
+import { MyInfoFactory } from '../../../services/myinfo/myinfo.factory'
+import { createReqMeta } from '../../../utils/request'
+import * as FormService from '../../form/form.service'
+import { SpcpFactory } from '../../spcp/spcp.factory'
+import {
+  createCorppassParsedResponses,
+  createSingpassParsedResponses,
+} from '../../spcp/spcp.util'
+import * as SubmissionService from '../submission.service'
+
+import * as EmailSubmissionService from './email-submission.service'
+import {
+  mapAttachmentsFromResponses,
+  mapRouteError,
+} from './email-submission.util'
+
+const logger = createLoggerWithLabel(module)
+
+export const handleEmailSubmission: RequestHandler<
+  { formId: string },
+  { message: string; submissionId?: string; spcpSubmissionFailure?: boolean },
+  { responses: FieldResponse[]; isPreview: boolean },
+  { captchaResponse?: unknown }
+> = async (req, res) => {
+  const { formId } = req.params
+  const logMeta = {
+    action: 'handleEmailSubmission',
+    ...createReqMeta(req as Request),
+    formId,
+  }
+  // Retrieve form
+  const formResult = await FormService.retrieveFullFormById(
+    formId,
+  ).andThen((form) => EmailSubmissionService.checkFormIsEmailMode(form))
+  if (formResult.isErr()) {
+    logger.error({
+      message: 'Error while retrieving form from database',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+  const form = formResult.value
+
+  // Check that form is public
+  const formPublicResult = FormService.isFormPublic(form)
+  if (formPublicResult.isErr()) {
+    logger.warn({
+      message: 'Attempt to submit non-public form',
+      meta: logMeta,
+      error: formPublicResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formPublicResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  // Check captcha
+  if (form.hasCaptcha) {
+    const captchaResult = await CaptchaFactory.verifyCaptchaResponse(
+      req.query.captchaResponse,
+      req.connection.remoteAddress,
```

Comment:
i think this is a bug and we should be using our `getRequestIp` helper function instead, as the IP here would be from the load balancer/cloudflare proxy.